app/controllers/declaration/AdditionalFiscalReferencesRemoveController.scala

```diff
@@ -0,0 +1,105 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.declaration
+
+import controllers.actions.{AuthAction, JourneyAction}
+import controllers.navigation.Navigator
+import controllers.util.MultipleItemsHelper.remove
+import forms.common.YesNoAnswer
+import forms.common.YesNoAnswer.YesNoAnswers
+import forms.declaration.{AdditionalFiscalReference, AdditionalFiscalReferencesData, AdditionalInformation}
+import javax.inject.Inject
+import models.requests.JourneyRequest
+import models.{ExportsDeclaration, Mode}
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import services.cache.ExportsCacheService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.ListItem
+import views.html.declaration.fiscalInformation.additional_fiscal_references_remove
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AdditionalFiscalReferencesRemoveController @Inject()(
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  override val exportsCacheService: ExportsCacheService,
+  navigator: Navigator,
+  mcc: MessagesControllerComponents,
+  removePage: additional_fiscal_references_remove
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport with ModelCacheable with SubmissionErrors {
+
+  def displayPage(mode: Mode, itemId: String, id: String): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
+    findAdditionalFiscalReference(itemId, id) match {
+      case Some(reference) => Ok(removePage(mode, itemId, id, reference, removeYesNoForm.withSubmissionErrors()))
+      case _               => returnToSummary(mode, itemId)
+    }
+  }
+
+  def submitForm(mode: Mode, itemId: String, id: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    findAdditionalFiscalReference(itemId, id) match {
+      case Some(reference) =>
+        removeYesNoForm
+          .bindFromRequest()
+          .fold(
+            (formWithErrors: Form[YesNoAnswer]) => Future.successful(BadRequest(removePage(mode, itemId, id, reference, formWithErrors))),
+            formData => {
+              formData.answer match {
+                case YesNoAnswers.yes =>
+                  removeAdditionalFiscalReference(itemId, reference)
+                    .map(declaration => afterRemove(mode, itemId, declaration))
+                case YesNoAnswers.no =>
+                  Future.successful(returnToSummary(mode, itemId))
+              }
+            }
+          )
+      case _ => Future.successful(returnToSummary(mode, itemId))
+    }
+
+  }
+
+  private def removeYesNoForm: Form[YesNoAnswer] = YesNoAnswer.form(errorKey = "declaration.additionalFiscalReferences.remove.empty")
```

Comment:
Nit: I'd suggest changing this name to `removalYesNoForm`. The method returns the form, but the name suggests something different when you don't have a context.